controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -41,22 +41,19 @@ func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCI
 }
 
 func (r *DSCInitializationReconciler) removeServiceMesh(instance *dsciv1.DSCInitialization) error {
-	// on condition of Managed, do not handle Removed when set to Removed it trigger DSCI reconcile to cleanup
-	if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
-		serviceMeshInitializer := feature.ClusterFeaturesInitializer(instance, configureServiceMeshFeatures())
-		if err := serviceMeshInitializer.Prepare(); err != nil {
-			r.Log.Error(err, "failed configuring service mesh resources")
-			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed configuring service mesh resources")
+	serviceMeshInitializer := feature.ClusterFeaturesInitializer(instance, configureServiceMeshFeatures())
```

Comment:
I am not sure what was the intent of this comment:

`on condition of Managed, do not handle Removed when set to Removed it trigger DSCI reconcile to cleanup`

@israel-hdez can you shed some light? :)

Reply:
i am the one put the comments there ( @israel-hdez is innocent) and i could not think what exactly that meant.
the only guess is: 
- on the Managed/Addon cluster, we do not explicitly delete SMCP CR even we set the serviermesh.state to "Removed". 
- the cleanup of SMCP CR should be done by deleting DSCI or even uninstall operator

but again, why?

Reply:
I''m now confused about what we wanted to do  here.

What I can say is that this was dead code and, with this change, it is no longer dead code.

The callee checks the same field for `Removed` and here we check for `Managed`. So, this code would run only if `instance.Spec.ServiceMesh.ManagementState` is both `Removed` and `Managed` at the same time...

We may need to go back in time to understand what we wanted to do  here...

Reply:
another possibility with my comments added in Dec is:
when this block first introduced, we do not want to delete SMCP CR even we set .dsci.servicemesh.state to Removed, the deletion should only happen when DSCI CR is deleted.
(have to say, Managed could have 2 meanings: Managed Cluster or value "Managed" on state)

Reply:
I called out @israel-hdez not to explain the comment but to have his opinion on what we should be doing here. I should have been more accurate with the ask, sorry for that!

I think there's a confusion on what `Removed` should mean and I am not sure I am exposed well enough to understand how our users are operating. My line of thinking was - you don't need service mesh anymore - we remove it with all the consequences. Generally speaking I see it as unlikely that users will be flipping the flag back and forth so it is really a deliberate decision in my eyes. Perhaps we can validate this statement with help of @danielezonca?

One thing which is definitely something I lack understanding of is the whole `addon` scenario.

> on the Managed/Addon cluster, we do not explicitly delete SMCP CR even we set the serviermesh.state to "Removed".

@zdtsw if it is not done explicitly - how is it performed and why?

Reply:
In general we are not sure that SMCP has not been changed by the user after the creation. This is probably the reason why the behaviour is to don't remove it.
I think we need to introduce `Unmanaged` state as soon as possible to distinguish the two different scenarios.

Reply:
correct.

logic of "Unmanaged" has been included in the code: https://github.com/opendatahub-io/opendatahub-operator/blob/db3bf7b89e8eac79ea58d8ca5442e34cc7b2e1dc/controllers/dscinitialization/servicemesh_setup.go#L31
to different from "Removed"  https://github.com/opendatahub-io/opendatahub-operator/blob/db3bf7b89e8eac79ea58d8ca5442e34cc7b2e1dc/controllers/dscinitialization/servicemesh_setup.go#L33

Reply:
> I called out @israel-hdez not to explain the comment but to have his opinion on what we should be doing here. I should have been more accurate with the ask, sorry for that!
> 
> I think there's a confusion on what `Removed` should mean and I am not sure I am exposed well enough to understand how our users are operating. My line of thinking was - you don't need service mesh anymore - we remove it with all the consequences. Generally speaking I see it as unlikely that users will be flipping the flag back and forth so it is really a deliberate decision in my eyes. Perhaps we can validate this statement with help of @danielezonca?
> 
> One thing which is definitely something I lack understanding of is the whole `addon` scenario.
> 
> > on the Managed/Addon cluster, we do not explicitly delete SMCP CR even we set the serviermesh.state to "Removed".
> 
> @zdtsw if it is not done explicitly - how is it performed and why?

This is only one guess when i tried to justify what exactly the "comments" means.
I can only vaguely remember in Addon/Managed Cluster SRE are responsibility for certain operation, but I do not know if this is one of them or i am just gibberish.

Reply:
> In general we are not sure that SMCP has not been changed by the user after the creation. This is probably the reason why the behaviour is to don't remove it.

This is the reasoning I have stamped in my brain. With this in mind, I'd think that:
* It is right that when flipping from `Managed` to `Removed`, the code is doing right on leaving untouched OSSM.
* We may need to fix the deletion of the `DSCI` to also leave untouched OSSM.

However, with Authorino coming, we may need to do some clean-up in either case.